src/administrative-sdk/progress/progress-controller.js

```diff
@@ -0,0 +1,43 @@
+import Progress from './progress';
+
+/**
+ * Controller class for the Progress model.
+ * @private
+ */
+export default class ProgressController {
+  /**
+   * @param {Connection} connection - Object to use for making a connection to the REST API and Websocket server.
+   */
+  constructor(connection) {
+    /**
+     * Object to use for making a connection to the REST API and Websocket server.
+     * @type {Connection}
+     */
+    this._connection = connection;
+  }
+
+  /**
+   * Get Progress corresponding to a {@Link Category} for the current active {@Link User} derived from
+   * the OAuth2 scope. If a user is allowed to see the progress of multiple users (depending of its
+   * {@Link Role} and permissions) then more Progress objects could be returned.
+   *
+   * Progress is always being fetched for a single category, identified with a categoryId.
+   *
+   * @param {string} categoryId - Specify a Category identifier.
+   * @returns {Promise.<Progress[]>} Array of Progress.
```

Comment:
Well.. in this case not. A Promise will be returned. And the resolved result will then be an Array with Progress objects. So Promise.<Progress[] is OK here.

As a side note, I looked at other controllers how it was done..